src/Microsoft.Sbom.Parsers.Spdx22SbomParser/SPDXParser.cs

```diff
@@ -159,8 +168,68 @@ ParserState MoveToNextState()
         }
     }
 
+    private ParserState ProcessInternalMetadataProperties()
+    {
+        ParserState internalParserState;
+        do
+        {
+            internalParserState = ProcessPropertyInternal();
+        }
+        while (internalParserState == ParserState.INTERNAL_METADATA);
+
+        return internalParserState;
+
+        ParserState ProcessPropertyInternal()
+        {
+            var reader = new Utf8JsonReader(buffer, isFinalBlock: isFinalBlock, readerState);
+            
+            switch (currentRootPropertyName)
+            {
+                case Constants.SPDXVersionHeaderName:
+                    metadata.SpdxVersion = nextTokenString;
```

Comment:
Why do this instead of `ParserUtils.GetString()` or whatever? Whatever the reason is we should codify it in a comment for future generations.